docs/design/supporting-multi-arch.md

```diff
@@ -0,0 +1,137 @@
+
+Otherwise both new labels should be specified and present, and bpfman will determine
+the endianness of the machine it's currently running on and ensure the correct
+bytecode is loaded.
+
+For example:
+
+```console
+cat Containerfile.bytecode 
+FROM scratch
+
+ARG PROGRAM_NAME
+ARG BPF_FUNCTION_NAME
+ARG EL_BYTECODE_FILE_PATH
+ARG EB_BYTECODE_FILE_PATH
+
+COPY  $EB_BYTECODE_FILE_PATH /
+COPY  $EL_BYTECODE_FILE_PATH /
+LABEL io.ebpf.file_eb $EB_BYTECODE_FILE_PATH
+LABEL io.ebpf.file_el $EB_BYTECODE_FILE_PATH
+LABEL io.ebpf.program_name $PROGRAM_NAME
+LABEL io.ebpf.bpf_function_name $BPF_FUNCTION_NAME
+
+docker build \
+ --build-arg PROGRAM_NAME=go_xdp_counter \
+ --build-arg BPF_FUNCTION_NAME=xdp_stats \
+ --build-arg PROGRAM_TYPE=xdp \
+ --build-arg EL_BYTECODE_FILE_PATH=bpf_bpfel.o \
+ --build-arg EB_BYTECODE_FILE_PATH=bpf_bpfeb.o \
+ -f Containerfile.bytecode \
+ ./examples/go-xdp-counter -t quay.io/astoycos/go_xdp_counter:latest
+
+[astoycos@nfvsdn-03 bpfman]$ skopeo inspect docker://quay.io/astoycos/go_xdp_counter_multi
+{
+    "Name": "quay.io/astoycos/go_xdp_counter_multi",
+    "Digest": "sha256:1b5d9bf983e8f636034a5b84ae6bd116288563bf21b4cc900c0550eb3e7120ca",
+    "RepoTags": [
+        "latest"
+    ],
+    "Created": "2024-05-16T13:40:45.819239992-04:00",
+    "DockerVersion": "",
+    "Labels": {
+        "io.ebpf.bpf_function_name": "xdp_stats",
+        "io.ebpf.file_eb": "bpf_bpfeb.o",
+        "io.ebpf.file_el": "bpf_bpfeb.o",
+        "io.ebpf.program_name": "go_xdp_counter"
+    },
+    "Architecture": "amd64",
+    "Os": "linux",
+    "Layers": [
+        "sha256:8184c4c42704a1a0717630af8caf2cc29ca0c32e966036f5cf75e0bbf0efd379",
+        "sha256:894e2011156116b36d6860f544a517bc7b29039f327c87e6fef010b3888b2527"
+    ],
+    "LayersData": [
+        {
+            "MIMEType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+            "Digest": "sha256:8184c4c42704a1a0717630af8caf2cc29ca0c32e966036f5cf75e0bbf0efd379",
+            "Size": 2642,
+            "Annotations": null
+        },
+        {
+            "MIMEType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+            "Digest": "sha256:894e2011156116b36d6860f544a517bc7b29039f327c87e6fef010b3888b2527",
+            "Size": 2656,
+            "Annotations": null
+        }
+    ],
+    "Env": [
+        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+    ]
+}
+```
+
+
+For most types of programs which don't rely on architecture specific kernel headers
+this single bytecode image will run on on any and all linux architectures, however
+there are cases where this won't work, and different header files need to be
+compiled into the program for various architectures. In this case bpfman will
+support the use of traditional [OCI "fat manifests"](https://github.com/opencontainers/image-spec/blob/main/manifest.md) where each image referenced in the manifest will include el and eb variants, although
```

Comment:
I'd argue that it would be easier just to support a single way of doing things - i.e multi-arch manifests.
This is simpler for the end user as they don't need to throw out what they had when their requirements inevitably change - they can just tweak the deployment appropriately.

Reply:
ACK I was trying to make thing a bit easier since building individual images for various architectures is a bit more of a pain, but if it's going to be a huge rat's nests we might as well compile separate pieces of bytecode for each arch.  Maybe we can make is super easy with a `bpfman build` command that could be cool.  Thanks sos much for the review Let me do a V2 of this